### PR TITLE
[1/N] Enable Wunused-result and Wunused-variable in torch targets

### PR DESCRIPTION
**Pull request by cyyever**
They are useful for checking results of function calls.

### CODE REVIEW

File: aten/src/ATen/native/NNPACK.cpp [hunk 1]
```diff
@@ -121,7 +121,11 @@ struct Workspace {
     constexpr size_t nnpack_memory_alignment_boundary = 64;
 
     // Won't work on Windows, but NNPACK doesn't support Windows either
-    posix_memalign(&buffer, nnpack_memory_alignment_boundary, size);
+    auto res = posix_memalign(&buffer, nnpack_memory_alignment_boundary, size);
```
Comment (Skylion007) [inline aten/src/ATen/native/NNPACK.cpp:124]: Since we are on C++17, can we use std::aligned_alloc now?

**Review by Skylion007 — APPROVED** [review]

File: aten/src/ATen/native/NNPACK.cpp (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/native/NNPACK.cpp:124]: I will try on another PR

**Pull request closed without merging**